vendor/wp-now/src/execute-wp-cli.ts

```diff
@@ -1,29 +1,27 @@
 import startWPNow from './wp-now';
 import { downloadWpCli } from './download';
 import getWpCliPath from './get-wp-cli-path';
-import getWpNowConfig, { WPNowMode } from './config';
+import getWpNowConfig, { WPNowMode, WPNowOptions } from './config';
 import { DEFAULT_PHP_VERSION, DEFAULT_WORDPRESS_VERSION } from './constants';
 import { phpVar } from '@php-wasm/util';
 
 /**
  * This is an unstable API. Multiple wp-cli commands may not work due to a current limitation on php-wasm and pthreads.
  */
-export async function executeWPCli ( projectPath: string, args: string[] ): Promise<{ stdout: string; stderr: string; }> {
+export async function executeWPCli( projectPath: string, args: string[], forcedWPNowOptions?: WPNowOptions ): Promise<{ stdout: string; stderr: string; }> {
 	await downloadWpCli();
-	const options = await getWpNowConfig({
+	let options = await getWpNowConfig({
 		php: DEFAULT_PHP_VERSION,
 		wp: DEFAULT_WORDPRESS_VERSION,
 		path: projectPath,
 	});
-	let mode = options.mode;
-	if (options.mode !== WPNowMode.WORDPRESS) {
-		// In case of not having the site projectPath,
-		// we use index mode to avoid other logic like downloading WordPress
-		mode = WPNowMode.INDEX;
+	options.mode = options.mode !== WPNowMode.WORDPRESS ? WPNowMode.INDEX : options.mode;
+	if (forcedWPNowOptions) {
+		options = { ...options, ...forcedWPNowOptions };
 	}
-	const { phpInstances, options: wpNowOptions } = await startWPNow({
+
+	const { phpInstances } = await startWPNow({
```

Comment:
In https://github.com/Automattic/studio/pull/213, we decided to avoid starting the site for running `wp-cli` commands. The rational behind this is mainly for performance reasons, as starting a site can take several seconds.

Reply:
@kozer do you foresee any conflict with that change?

Reply:
No, I don't. However, adding index, does the same thing ( it stops after initiating the PHP instances ).
I like your solution better, and when things get merged, I will switch! :smile:

Reply:
Updated the PR!! Now I use your solution instead! Thank you @fluiddot